core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -170,7 +170,7 @@ public InstallState(@Nonnull String name, boolean isSetupComplete) {
     public void initializeState() {
     }
     
-    public Object readResolve() {
+    private Object readResolve() {
```

Comment:
Not a final class. Also it's an extension point, which may be potentially doing some migrations in custom implementations from plugins 🐛